First commits

jsf9k opened the pull request:
## 🗣 Description ##

This pull request contains the initial commits for an Ansible role that modifies OpenSSH to allow connections with RSA keys.

## 💭 Motivation and context ##

This Ansible role is necessary for [the latest Kali AMI](), since the base AMI has OpenSSH 9.  The `sshd` in that version disallows RSA connections by default, and that is the only key type currently supported by Guacamole.

## 🧪 Testing ##

Tested as part of cisagov/kali-packer#111.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

jsf9k committed:
Update README and role metadata from skeleton defaults

jsf9k committed:
Modify the role to change the OpenSSH config to support RSA keys

dav3r reviewed (approved):
Looks great! 🦊

mcdonnnj commented on molecule/default/tests/test_default.py:
```diff
@@ -12,7 +12,12 @@
-    """Run a dummy test, just to show what one would look like."""
-    assert x
+@pytest.mark.parametrize("f", ["/etc/ssh/sshd_config.d/99-allow-rsakeys.conf"])
+def test_files(host, f):
```
Just as a thought should this verify that the two keywords are there?

jsf9k replied:
Please see commit 7571ac6, in which I verify that the correct version of the `PubkeyAccepted*` keyword is there.  I don't think it makes sense to check for the other keyword, since that would simply be testing the `ansible.builtin.copy` module.

jsf9k replied:
Can this conversation be resolved?

mcdonnnj commented on tasks/main.yml:
```diff
@@ -1,2 +1,29 @@
+  ansible.builtin.copy:
+    content: |
+      HostKeyAlgorithms +ssh-rsa
+      PubkeyAcceptedAlgorithms +ssh-rsa
```
I believe this command requires OpenSSH 8.5 or newer. Are we still ok to leave the supported platforms as-is when, for example, Debian Stretch only has 7.4 available?

jsf9k replied:
Please see commit 7571ac6.

jsf9k replied:
Can this conversation be resolved?

mcdonnnj replied:
Sorry, I hit resolve in the GitHub app when this notification came in but yet here this conversation sits, unresolved.

mcdonnnj reviewed (approved):
Straightforward. I have two questions that I wanted to voice, but they do not hinder my approval.

jsf9k committed:
Pre-install OpenSSH client (ssh)

jsf9k committed:
Modify ssh config fragment according to what version of OpenSSH is installed

The tasks.yml file is modified to determine what version of OpenSSH is
installed, then install an OpenSSH configuration fragment that is
correct for that version.

The test code is modified to check that the correct version of the
OpenSSH configuration fragment is installed on each platform.

dav3r commented on molecule/default/tests/test_default.py:
```diff
@@ -12,7 +12,44 @@
+
+
+def test_file_content(host):
+    """Verify that all file contains the expected content."""
```
```suggestion
    """Verify that all files contain the expected content."""
```

mcdonnnj replied:
Would this instead be the following because we only check a single file? Alternatively something more descriptive than just "the file" would be helpful.

```suggestion
    """Verify that the file contains the expected content."""
```

dav3r replied:
Yup, @mcdonnnj's suggestion makes sense and is probably what @jsf9k intended.

jsf9k replied:
Please see commit 9145465.

dav3r reviewed (approved):
Just a minor grammatical nitpick, otherwise all good 👍

mcdonnnj commented on molecule/default/tests/test_default.py:
```diff
@@ -12,7 +12,44 @@
+            assert ff.contains("^PubkeyAcceptedKeyTypes")
+        elif host.system_info.codename in ["bookworm"]:
+            # OpenSSH 8.5+
+            assert ff.contains("^PubkeyAcceptedAlgorithms")
```
I do not believe there is an Ubuntu Bookworm 😅

```suggestion
```

jsf9k replied:
Please see commit a6b3e42.

mcdonnnj reviewed (approved):
Thanks for taking my previous comments under consideration. This is a nice improvement and I only have a small item to request. Strong work 💪💪💪

jsf9k committed:
Fix grammar

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Fix copy-and-paste error

There is no Ubuntu Bookworm.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>
